### PR TITLE
test(starknet_mempool_p2p): test tx received from p2p reach mempool

### PR DESCRIPTION
**Pull request by AlonLStarkWare**
- chore: derive Hash on RpcTransaction
- chore: changing closure Fn requirements on integration test utils
- test(mempool_p2p): test received rpc transactions are being broadcasted
- test(mempool_p2p): test tx received from p2p reach mempool

### CODE REVIEW

File: crates/tests_integration/tests/mempool_p2p_flow_test.rs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    // Wait for server to spin up and for p2p to discover other peer.
+    // TODO(Gilad): Replace with a persistent Client with a built-in retry to protect against CI
+    // flakiness.
+    tokio::time::sleep(std::time::Duration::from_millis(1000)).await;
+
+    let mut expected_txs = HashSet::new();
+
+    let _tx_hashes = run_integration_test_scenario(tx_generator, &mut |tx: RpcTransaction| {
+        expected_txs.insert(tx.clone());
+        ready(TransactionHash::default()) // using the default value becuase we don't use the hash anyways.
```
[the remark this answers is not in the document]
Comment (AlonLStarkWare) [reply crates/tests_integration/tests/mempool_p2p_flow_test.rs]: 👍

**Review by AlonLStarkWare — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: 0 of 12 files reviewed, 1 unresolved discussion (waiting on @graphite-app[bot] and @ShahakShama)

**Review by ShahakShama — CHANGES_REQUESTED** [review]
> Reviewed 1 of 8 files at r4, 4 of 5 files at r6, 1 of 7 files at r8, 6 of 6 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: all files reviewed, 5 unresolved discussions (waiting on @AlonLStarkWare)

___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 136 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBe6aGPCIIFAAlR2Zkk:-OBe6aGPCIIFAAlR2Zkl:b-gys401) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L136)):*
> ```XML
> #[tokio::test]
> async fn test_mempool_receives_tx_from_other_peer(tx_generator: MultiAccountTransactionGenerator) {
>     let handle = Handle::current();
> ```

Remove code dup with other test by defining a setup function in this module (it can be private)
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 205 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBeZVxo1KFH1QpoHDUn:-OBeZVxo1KFH1QpoHDUo:bg320e5) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L205)):*
> ```XML
>     }
> 
>     // waiting for the tx to be received (TODO: figure out a better solution)
> ```

You can iteratively call get_txs with a sleep of much lower time until you get expected_txs.len() txs
Fail if you passed x iterations (calculate x so that the time it will take is 2 seconds)
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 208 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBeZoW04WFNM_ZVeuTp:-OBeZoW04WFNM_ZVeuTq:bg6q9hy) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L208)):*
> ```XML
>     tokio::time::sleep(std::time::Duration::from_millis(2000)).await;
> 
>     let received_tx = mempool_client.get_txs(expected_txs.len()).await.unwrap();
> ```

received_txs
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 213 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBe_-9x2To2nGtKbP2z:-OBe_-9x2To2nGtKbP3-:b-fon6vc) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L213)):*
> ```XML
>     assert_eq!(received_tx.len(), expected_txs.len());
> 
>     for tx in received_tx.iter() {
> ```

No need to write iter here. Surprised clippy and graphite-bot didn't say anything
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 214 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBeZsvG0BzW8XtCbCEV:-OBeZsvG0BzW8XtCbCEW:b-msuik3) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L214)):*
> ```XML
> 
>     for tx in received_tx.iter() {
>         let converted_tx: RpcTransaction = match tx {
> ```

Add TODO to change mempool to store RpcTransaction

**Review by AlonLStarkWare — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: all files reviewed, 5 unresolved discussions (waiting on @ShahakShama)

___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 136 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBe6aGPCIIFAAlR2Zkk:-OBzokdcBM1P04AMNh6y:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L136)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Remove code dup with other test by defining a setup function in this module (it can be private)
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 205 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBeZVxo1KFH1QpoHDUn:-OC2jpN97y3yurt0Dz1W:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L205)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

You can iteratively call get_txs with a sleep of much lower time until you get expected_txs.len() txs
Fail if you passed x iterations (calculate x so that the time it will take is 2 seconds)
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 208 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBeZoW04WFNM_ZVeuTp:-OC2joat1FiP8qYp_106:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L208)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

received_txs
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 213 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBe_-9x2To2nGtKbP2z:-OC2jmIdDAZT-G1_D-MX:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L213)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

No need to write iter here. Surprised clippy and graphite-bot didn't say anything
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 214 at r9](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OBeZsvG0BzW8XtCbCEV:-OC2jlv1CfXZMK_6mlRH:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/0b0ec55ac9c6e7235a8491f7ba7d258f762d5eee/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L214)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Add TODO to change mempool to store RpcTransaction
</blockquote></details>

Done.

**Review by ShahakShama — CHANGES_REQUESTED** [review]
> Reviewed 4 of 6 files at r10, 1 of 1 files at r11, 1 of 1 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: all files reviewed, 6 unresolved discussions (waiting on @AlonLStarkWare)

___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 78 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDiGjnAbdlJ94IR1n_:-OCDiGjnAbdlJ94IR1na:bstxc3f) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L78)):*
> ```XML
>     };
> 
>     let chain_info = create_chain_info();
> ```

This line is a duplicate. Surprised clippy passed
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 161 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDj4PzDPo0FGw5VRxq:-OCDj4PzDPo0FGw5VRxr:bo6xvku) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L161)):*
> ```XML
>     })
>     .await;
>     for tx in expected_txs.iter() {
> ```

There's still iter here. Use `&expected_txs` instead
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 169 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDk8fW8ht6_5T75GMO:-OCDk8fW8ht6_5T75GMP:b5inmhc) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L169)):*
> ```XML
>     }
> 
>     let mut received_transactions: Vec<AccountTransaction> = vec![];
> ```

Either rename expected_txs to expected_transactions, or rename this to received_txs
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 170 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDjwCy6_rOMh72c061:-OCDjwCy6_rOMh72c062:bti5o4i) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L170)):*
> ```XML
> 
>     let mut received_transactions: Vec<AccountTransaction> = vec![];
>     for _ in 0..20 {
> ```

Make 20 and 100 constants
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 175 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDk3EiERyEyI2E_OzR:-OCDk3EiERyEyI2E_OzS:b-1lbmmz) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L175)):*
> ```XML
>         }
>         received_transactions
>             .append(&mut mempool_client.get_txs(expected_txs.len()).await.unwrap());
> ```

This should be expected_txs.len() - received_transactions.len()
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 41 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDhxdn6vCBU1nGGJTt:-OCDhxdn6vCBU1nGGJTu:b-1dcon) ([raw file](https://github.com/starkware-libs/sequencer/blob/4f066981ebebcc5150b8cce27008fd230656fe3c/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L41)):*
> ```XML
> }
> 
> // TODO: remove code duplication with FlowTestSetup
> ```

Keep this TODO on top of the setup function. It's still not done

**Review by AlonLStarkWare — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: 10 of 12 files reviewed, 6 unresolved discussions (waiting on @ShahakShama)

___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 41 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDhxdn6vCBU1nGGJTt:-OD0gjdp5by6q_sOmwWN:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/4f066981ebebcc5150b8cce27008fd230656fe3c/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L41)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Keep this TODO on top of the setup function. It's still not done
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 78 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDiGjnAbdlJ94IR1n_:-OD0dF1JEH-yo6yFA4I4:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L78)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

This line is a duplicate. Surprised clippy passed
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 161 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDj4PzDPo0FGw5VRxq:-OD0fkJkDBPZGtYXUx34:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L161)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

There's still iter here. Use `&expected_txs` instead
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 169 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDk8fW8ht6_5T75GMO:-OD0dWFxE1mM0aBsFlCV:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L169)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Either rename expected_txs to expected_transactions, or rename this to received_txs
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 170 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDjwCy6_rOMh72c061:-OD0gqO427SZm6u60XVM:b-z7j79e) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L170)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Make 20 and 100 constants
</blockquote></details>

Made 100 a constant and derived 20 by making sure we stop after 2 seconds.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 175 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDk3EiERyEyI2E_OzR:-OD0eXo_1yPvdz_P3KFL:b-psv1f6) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L175)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

This should be expected_txs.len() - received_transactions.len()
</blockquote></details>

This shouldn't matter because get_txs will get up to n_txs txs, but I changed it either way.

**Review by ShahakShama — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r15, 1 of 1 files at r17, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: all files reviewed, 3 unresolved discussions (waiting on @AlonLStarkWare)

___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 41 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDhxdn6vCBU1nGGJTt:-ODLZs_mB7-yC0C9LuDA:b-1fqqup) ([raw file](https://github.com/starkware-libs/sequencer/blob/4f066981ebebcc5150b8cce27008fd230656fe3c/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L41)):*
<details><summary><i>Previously, AlonLStarkWare (Alon-Lukatch-Starkware) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

The TODO was "// TODO: remove code duplication with FlowTestSetup"
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 170 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDjwCy6_rOMh72c061:-ODL_8dJCuecQ0Rc-dv4:b-aoknue) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L170)):*
<details><summary><i>Previously, AlonLStarkWare (Alon-Lukatch-Starkware) wrote…</i></summary><blockquote>

Made 100 a constant and derived 20 by making sure we stop after 2 seconds.
</blockquote></details>

Extract 2000 to a constant
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 175 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDk3EiERyEyI2E_OzR:-ODL_KorCwMhL2z3bNFN:b-4lu4mw) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L175)):*
<details><summary><i>Previously, AlonLStarkWare (Alon-Lukatch-Starkware) wrote…</i></summary><blockquote>

This shouldn't matter because get_txs will get up to n_txs txs, but I changed it either way.
</blockquote></details>

You're right, we need to make sure there are just expected_txs.len() and not more. change this to expected_txs.len() - received_txs.len() + 1 and add a comment that this way you verify there are no extra txs

**Review by AlonLStarkWare — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: all files reviewed, 3 unresolved discussions (waiting on @graphite-app[bot] and @ShahakShama)

___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 41 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDhxdn6vCBU1nGGJTt:-ODLlGs28EFW9KeQep38:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/4f066981ebebcc5150b8cce27008fd230656fe3c/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L41)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

The TODO was "// TODO: remove code duplication with FlowTestSetup"
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 170 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDjwCy6_rOMh72c061:-ODLmfnK2LT03JoNAkEf:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L170)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Extract 2000 to a constant
</blockquote></details>

Done.
___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 175 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OCDk3EiERyEyI2E_OzR:-ODLnDo05Yib_1yMIzHY:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/94f7763eeacababc2a573a6502d49c53feccd666/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L175)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

You're right, we need to make sure there are just expected_txs.len() and not more. change this to expected_txs.len() - received_txs.len() + 1 and add a comment that this way you verify there are no extra txs
</blockquote></details>

Done.

**Review by ShahakShama — APPROVED** [review]
> Reviewed 1 of 1 files at r20, 1 of 1 files at r22, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: all files reviewed, 1 unresolved discussion (waiting on @AlonLStarkWare)

___
*[`crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs` line 146 at r22](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-OD_z2be9VDtWPHXvIpA:-OD_z2be9VDtWPHXvIpB:b-5b5a7t) ([raw file](https://github.com/starkware-libs/sequencer/blob/07be965f79a85056567accca61c3292306bd4c7e/crates/starknet_integration_tests/tests/mempool_p2p_flow_test.rs#L146)):*
> ```XML
> ) {
>     const RECEIVED_TX_POLL_INTERVAL: u64 = 100; // milliseconds between calls to read received txs from the broadcast channel
>     const TXS_RETRIVAL_TIMEOUT: u64 = 2000; // max milliseconds spent polling the received txs before timing out 
> ```

you have a trailing space at the end of the sentence

**Review by ShahakShama — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r23, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990#-:-ODaEtra3m-rrgCAjA6n:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @AlonLStarkWare)

**Review by AlonLStarkWare — COMMENTED** [review]
> Reviewed 1 of 8 files at r4, 4 of 5 files at r6, 1 of 6 files at r9, 4 of 6 files at r10, 1 of 1 files at r20, 1 of 1 files at r23, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @AlonLStarkWare)

**Review by AlonLStarkWare — COMMENTED** [review]
> Reviewed 1 of 1 files at r25, 1 of 1 files at r27, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @AlonLStarkWare)

**Review by ShahakShama — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1990)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @AlonLStarkWare)

**Pull request merged**